apps/cowswap-frontend/src/modules/trade/hooks/useTradeRouteContext.ts

```diff
@@ -14,6 +14,7 @@ export function useTradeRouteContext(): TradeUrlParams {
   const prevContextRef = useRef<TradeUrlParams>()
 
   const { orderKind, inputCurrencyAmount, outputCurrencyAmount } = derivedState || {}
+  const isStateExists = !!state
```

Comment:
nit, sounds like bad English: what about `hasState`?

Reply:
Yep, it is a bad English :) 
Sorry, renamed it